setup.py

```diff
@@ -4,6 +4,37 @@
 
 from setuptools import find_packages, setup
 
+# Fail with explicit error if trying to install Django 2 on Python 2, and
+# advise on how to avoid this.
+if sys.version_info < (3,):
+    sys.stderr.write(
+        """
+
+========
+WARNING!
+========
+
+You are running Django >= 2 on Python 2, but it requires Python 3.
+
+Unfortunately Django 2.0 and above are not compatible with Python 2,
+but you still ended up trying to install this version.  This should
+not have happened.  Make sure you have pip >= 9.0 to avoid this kind
+of issue, as well as setuptools >= 24.2:
+
+    $ pip install pip setuptools --upgrade
```

Comment:
The usual convention is to put flags before positional arguments. Move `--upgrade` before the packages.

Reply:
Also, doesn't this fail on Windows? Best use `python -m pip ...` to be safe...